libpldmresponder/fru.cpp

```diff
@@ -22,118 +26,117 @@ namespace responder
 
 constexpr auto root = "/xyz/openbmc_project/inventory/";
 
-std::optional<pldm_entity>
-    FruImpl::getEntityByObjectPath(const dbus::InterfaceMap& intfMaps)
+pldm_entity FruImpl::getEntityByObjectPath(const dbus::ObjectValueTree& objects,
+                                           const std::string& path)
 {
-    for (const auto& intfMap : intfMaps)
+    pldm_entity entity{};
+
+    auto iter = objects.find(path);
+    if (iter == objects.end())
+    {
+        return entity;
+    }
+
+    for (const auto& intfMap : iter->second)
     {
         try
         {
-            pldm_entity entity{};
             entity.entity_type = parser.getEntityType(intfMap.first);
-            return entity;
+            entity.entity_instance_num = 1;
+            break;
         }
         catch (const std::exception&)
         {
             continue;
         }
     }
 
-    return std::nullopt;
+    return entity;
 }
 
 void FruImpl::updateAssociationTree(const dbus::ObjectValueTree& objects,
                                     const std::string& path)
```

Comment:
why is there a need to change this function ? in CM and hotplug - this function has been already improved and merged in upstream.